vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/NetworkConditionsSteps.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.mobileapp.steps;
+
+import org.apache.commons.lang3.Validate;
+import org.jbehave.core.annotations.When;
+import org.vividus.mobileapp.action.NetworkActions;
+import org.vividus.mobileapp.action.NetworkActions.Mode;
+import org.vividus.mobileapp.action.NetworkActions.NetworkToggle;
+import org.vividus.selenium.manager.GenericWebDriverManager;
+
+public final class NetworkConditionsSteps
+{
+    private final NetworkActions networkActions;
+    private final GenericWebDriverManager genericWebDriverManager;
+
+    private NetworkConditionsSteps(GenericWebDriverManager genericWebDriverManager, NetworkActions networkActions)
+    {
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.networkActions = networkActions;
+    }
+
+    /**
+     * Turn <b>toggle</b> <b>mode</b> network connection
+     * @param toggle switch ON or OFF mode
+     * @param mode is network condition to be executed
+     * <p>The actions performed by the step for iOS:</p>
+     * <ul>
+     * <li>activate Preferences app</li>
+     * <li>open the selected mode</li>
+     * <li>switch toggle if necessary</li>
+     * </ul>
+     * <p>The actions performed by the step for Android:</p>
+     * <ul>
+     * <li>Initializes connection state builder and sets selected mode to disabled/enabled state if necessary</li>
+     * </ul>
```

Comment:
general info about method should go first, then parameters are, and then info about exceptions is placed